mock.go

```diff
@@ -42,13 +43,25 @@ func (m mockOp) Run() error {
 	return nil
 }
 
+func (m mockOp) RunWithContext(ctx context.Context) error {
+	return m.WithOptions(Options{Context: ctx}).Run()
+}
+
+func (m mockOp) Options() Options {
+	return m.options
+}
+
 func (m mockOp) WithOptions(opt Options) Op {
 	return mockOp{
 		options: opt,
 		funcs:   m.funcs,
 	}
 }
 
+func (m mockOp) RunAtomicallyWithContext(_ context.Context) error {
+	return m.Run()
```

Comment:
Should we be passing the context here?

Reply:
Also — super minor thing — as above could we order the WithContext func after the RunAtomically func?